### PR TITLE
#164047058 Users should receive descriptive error messages

### PR DESCRIPTION
**Pull request by JumaKahiga**
#### Description
___
Currently, users receives inbuilt Django error messages if they provide invalid registration details. Although these messages reduce tasks for the developer, they are usually generic, and thus the user may not understand what caused the error. It would be good to create custom error messages that resolve this issue.

#### Type of Change
___
[x] New Feature

#### How has this been tested
___
[x] Unit tests

#### Checklist
___
[x] I have added tests that prove my feature is working

#### Pivotal Tracker
___
[164047058](https://www.pivotaltracker.com/story/show/164047058)

### CODE REVIEW

**Comment by winniekariuki** [comment]
Good testing of edge cases.

File: authors/apps/authentication/tests/test_registration_error_messages.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+from django.shortcuts import reverse
+import json
+from rest_framework.test import APITestCase, APIClient
+from rest_framework.views import status 
+
+from ..models import User
+
+
+
```
Comment (Kibetchirchir) [inline authors/apps/authentication/tests/test_registration_error_messages.py:9]: Remove the blank space. Use PEP8

File: .gitignore [hunk 2]
```diff
@@ -97,3 +97,13 @@ db.sqlite3
 
 #ds_store
 .ds_store
+
+# Local Files
+.DS_Store
+authors/.DS_Store
+authors/apps/.DS_Store
+authors/apps/authentication/.DS_Store
+authors/apps/authentication/migrations/0001_initial.py
```
Comment (JoanNgatia) [inline .gitignore]: Remove this line from the `.gitignore` file and Just make sure that you do not stage the changes to the migrations file. Everything else looks okay

**Review by JoanNgatia — CHANGES_REQUESTED** [review]

**Comment by kelvinndmo** [comment]
Good work @JumaKahiga .

File: .gitignore (hunk 2 above)
Comment (JumaKahiga) [reply .gitignore]: Thank you for the feedback Joan. Fixed the file.

**Comment by mnswaleh** [comment]
Everything is good
